python/social_bot/tasks.py

```diff
@@ -598,18 +652,27 @@ def task_specific_observation(self, agent):
                 pose = np.concatenate((pose, obj_pos), axis=0)
 
         agent_pose = np.array(agent.get_pose()).flatten()
-        if self._use_egocentric_states:
+        if self._use_full_states or self._use_egocentric_states:
             yaw = agent_pose[5]
             # adds egocentric velocity input
             vx, vy, vz, a1, a2, a3 = np.array(agent.get_velocities()).flatten()
-            rvx, rvy = agent.get_egocentric_cord_2d(vx, vy, yaw)
-            obs = [rvx, rvy, vz, a1, a2, a3]
+            if self._use_egocentric_states:
+                rvx, rvy = agent.get_egocentric_cord_2d(vx, vy, yaw)
+            else:
+                rvx, rvy = vx, vy
+            obs = [rvx, rvy, vz, a1, a2, a3] + list(agent_pose)
             # adds objects' (goal's as well as distractions') egocentric
             # coordinates to observation
             while len(pose) > 1:
-                x = pose[0] - agent_pose[0]
-                y = pose[1] - agent_pose[1]
-                rotated_x, rotated_y = agent.get_egocentric_cord_2d(x, y, yaw)
+                x = pose[0]
+                y = pose[1]
+                if self._use_egocentric_states:
+                    x = pose[0] - agent_pose[0]
+                    y = pose[1] - agent_pose[1]
+                    rotated_x, rotated_y = agent.get_egocentric_cord_2d(
+                        x, y, yaw)
+                else:
+                    rotated_x, rotated_y = x, y
                 if self._egocentric_perception_range > 0:
```

Comment:
Maybe self._use_egocentric_states and elf._egocentric_perception_range > 0 ? 
If _use_egocentric_states is False, rotated_x is not rotated, the angle caculation below will not be correct.

Reply:
good catch.  done.